source/lac/petsc_vector_base.cc

```diff
@@ -210,6 +210,27 @@ namespace PETScWrappers
 
 
 
+  void
+  VectorBase::clear ()
+  {
+    if (attained_ownership)
+      {
+#if DEAL_II_PETSC_VERSION_LT(3,2,0)
+        const int ierr = VecDestroy (vector);
+#else
+        const int ierr = VecDestroy (&vector);
```

Comment:
I could go with this as well. It just seems unnecessary to have it *both* in the base and the derived classes.